rpcs3/Emu/NP/np_handler.cpp

```diff
@@ -983,8 +1052,39 @@ namespace np
 					}
 				}
 
+				auto presence_updates = rpcn->get_presence_updates();
+				if (basic_handler_registered)
+				{
+					for (auto& [npid, pr_info] : presence_updates)
+					{
+						basic_event to_add{};
+						strcpy_trunc(to_add.from.userId.handle.data, npid);
+						strcpy_trunc(to_add.from.name.data, npid);
+
+						if (std::memcmp(pr_info.pr_com_id.data, basic_handler.context.data, sizeof(pr_info.pr_com_id.data)) == 0)
+						{
+							to_add.event = SCE_NP_BASIC_EVENT_PRESENCE;
+							to_add.data = std::move(pr_info.pr_data);
+						}
+						else
+						{
+							if (basic_handler.context_sensitive && !pr_info.pr_com_id.data[0] == 0)
```

Comment:
```
[327/579] Building CXX object rpcs3/Emu/CMakeFiles/rpcs3_emu.dir/NP/np_handler.cpp.obj
C:/src/rpcs3/rpcs3/Emu/NP/np_handler.cpp:1076:47: warning: logical not is only applied to the left hand side of this comparison [-Wlogical-not-parentheses]
 1076 |                                                         if (basic_handler.context_sensitive && !pr_info.pr_com_id.data[0] == 0)
      |
 ^                          ~~
C:/src/rpcs3/rpcs3/Emu/NP/np_handler.cpp:1076:47: note: add parentheses after the '!' to evaluate the comparison first
 1076 |                                                         if (basic_handler.context_sensitive && !pr_info.pr_com_id.data[0] == 0)
      |
 ^
      |
  (                             )
C:/src/rpcs3/rpcs3/Emu/NP/np_handler.cpp:1076:47: note: add parentheses around left hand side expression to silence this warning
 1076 |                                                         if (basic_handler.context_sensitive && !pr_info.pr_com_id.data[0] == 0)
      |
 ^
      |
 (                         )
1 warning generated.
```

Reply:
Thanks, will be fixed in https://github.com/RPCS3/rpcs3/pull/15025